src/schema.graphql

```diff
@@ -205,18 +206,25 @@ type Query {
     phone: String @pattern(regexp: "^\\+[0-9]+$", policy: THROW)
   ): ID
   getLevels: [Int]
-  getLimits: Limits
+  getUserLimits: UserLimits
   getWalletFees: WalletFees
+  generate2fa: TwoFactorSecret
+  validate2fa(token: String!): Boolean
 }
 
-type WalletFees {
-  deposit: Float
+type UserLimits {
+  remainingTwoFactorLimit: Int
+  remainingOnUsLimit: Int
+  remainingWithdrawalLimit: Int
+}
+
+type TwoFactorSecret {
+  secret: String
+  uri: String
 }
 
-type Limits {
-  oldEnoughForWithdrawal: Int
-  withdrawal: Int
-  onUs: Int
+type WalletFees {
+  deposit: Float
 }
 
```

Comment:
Can we make the types fields not nullable?

Reply:
Done